### ISSUE TITLE
rake db:migrate fails after updating to stable

### ISSUE THREAD

**Issue by phleba** [loomio/loomio-deploy#85]
Hi, when updating from Loomio `v2.8.5` to `stable`, I get the following error when trying to migrate to the latest database schema

>  [13:24:33] xyz@xyz ~/compose/loomio $ docker-compose run app rake db:migrate
> Creating network "loomio_loomiomain" with the default driver
> Creating network "loomio_default" with the default driver
> Creating loomio-db    ... done
> Creating loomio-redis ... done
> Creating loomio_app_run ... done
> `Redis#exists(key)` will return an Integer by default in redis-rb 4.3. The option to explicitly disable this behaviour via 
> `Redis.exists_returns_integer` will be removed in 5.0. You should use `exists?` instead.
> Migrating to MigratePaperclipAttachments (20210826234109)
> == 20210826234109 MigratePaperclip
> Attachments: migrating ======================
> #<Group:0x0000563aca421200>
> #<Group:0x0000563aca9742c0>
> #<FormalGroup:0x0000563aca3256a8>
> #<FormalGroup:0x0000563ac9fa3958>
> #<GuestGroup:0x0000563ac9e60668>
> #<GuestGroup:0x0000563ac896c860>
> rake aborted!
> StandardError: An error has occurred, this and all later migrations canceled:
> PG::UndefinedTable: ERROR:  relation "templates" does not exist
> LINE 8:  WHERE a.attrelid = '"templates"'::regclass
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `exec'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `block (2 levels) in query'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:18:in `block in query'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract_adapter.rb:696:in `block (2 levels) in log'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract_adapter.rb:695:in `block in log'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract_adapter.rb:687:in `log'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:17:in `query'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/postgresql_adapter.rb:820:in `column_definitions'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:116:in `columns'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/schema_cache.rb:112:in `block in columns'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/schema_cache.rb:111:in `fetch'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/schema_cache.rb:111:in `columns'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/schema_cache.rb:120:in `block in columns_hash'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/schema_cache.rb:119:in `fetch'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/schema_cache.rb:119:in `columns_hash'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/model_schema.rb:553:in `load_schema!'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/attributes.rb:250:in `load_schema!'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/model_schema.rb:539:in `block in load_schema'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/model_schema.rb:536:in `synchronize'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/model_schema.rb:536:in `load_schema'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/model_schema.rb:397:in `columns'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/model_schema.rb:466:in `column_names'
> /loomio/db/migrate/20210826234109_migrate_paperclip_attachments.rb:7:in `block in up'
> /loomio/db/migrate/20210826234109_migrate_paperclip_attachments.rb:6:in `each'
> /loomio/db/migrate/20210826234109_migrate_paperclip_attachments.rb:6:in `up'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:870:in `public_send'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:870:in `exec_migration'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:851:in `block (2 levels) in migrate'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:850:in `block in migrate'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:462:in `with_connection'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:849:in `migrate'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1037:in `migrate'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1329:in `block in execute_migration_in_transaction'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1380:in `block in ddl_transaction'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `block in transaction'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:319:in `block in within_new_transaction'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
> /usr/local/bundle/gems/activesupport-6.1.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:317:in `within_new_transaction'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `transaction'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/transactions.rb:209:in `transaction'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1380:in `ddl_transaction'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1328:in `execute_migration_in_transaction'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1302:in `each'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1302:in `migrate_without_lock'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1251:in `block in migrate'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1401:in `block in with_advisory_lock'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1416:in `block in with_advisory_lock_connection'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:462:in `with_connection'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1416:in `with_advisory_lock_connection'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1397:in `with_advisory_lock'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1251:in `migrate'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1086:in `up'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/migration.rb:1061:in `migrate'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/tasks/database_tasks.rb:237:in `migrate'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/railties/databases.rake:92:in `block (3 levels) in <main>'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/railties/databases.rake:90:in `each'
> /usr/local/bundle/gems/activerecord-6.1.4.1/lib/active_record/railties/databases.rake:90:in `block (2 levels) in <main>'
> /usr/local/bundle/gems/sentry-ruby-core-4.8.1/lib/sentry/rake.rb:24:in `execute'
> /usr/local/bundle/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
> Caused by:
> ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "templates" does not exist
> LINE 8:  WHERE a.attrelid = '"templates"'::regclass`

Any ideas? Thanks a lot!

Edit: I will try to update in smaller steps later, let you know what happens then.

**Comment by phleba** [comment]
Updating in smaller steps worked propably due to migration from paperclips to activestorage in `v2.9.0`

**Comment by robguthrie** [comment]
Oh, glad you found a solution! Nice job.

**Comment by eternaltyro** [comment]
How do I update in smaller steps?

**Comment by robguthrie** [comment]
Hi @eternaltyro please post your error message and I'll see what I can do to help

**Comment by eternaltyro** [comment]
@robguthrie sorry about not posting an update earlier. I was trying to update from a really old version. I solved the problem after reading the note in the install documentation that mentioned that I should move from one stable milestone version to the next instead of trying to go all the way to the latest version. 

> Updating old versions of Loomio
> 
> Please upgrade through the following versions to ensure that migrations work v2.4.2 v2.8.8 v2.11.13

So I modified the compose file to place `v2.11.13` as the image tag, ran the migration, then made the tag `latest` and ran the migration again and everything worked. Thank you for the help, anyway and thank you for making this awesome software.
